src/Workspaces/CSharp/Portable/Formatting/Rules/NewLineUserSettingFormattingRule.cs

```diff
@@ -116,6 +116,15 @@ public override AdjustSpacesOperation GetAdjustSpacesOperation(SyntaxToken previ
                 }
             }
 
+            // * { - in the local function context
+            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) && currentTokenParentParent != null && currentTokenParentParent.IsKind(SyntaxKind.LocalFunctionStatement))
```

Comment:
Yes.  It is an expected roslyn pattern that IsKind and IsParentKind will do null checks, and thus you can elide the null check here.  Sneaky, but idiomatic in our codebase now.